be/src/olap/memtable.cpp

```diff
@@ -604,12 +611,87 @@ void MemTable::shrink_memtable_by_agg() {
     }
 }
 
-bool MemTable::need_flush() const {
+bool MemTable::need_flush(const vectorized::Block* block, const DorisVector<uint32_t>& row_idxs,
+                          PartialUpdateInfo* _partial_update_info) const {
     DBUG_EXECUTE_IF("MemTable.need_flush", { return true; });
     auto max_size = config::write_buffer_size;
     if (_partial_update_mode == UniqueKeyUpdateModePB::UPDATE_FIXED_COLUMNS) {
-        auto update_columns_size = _num_columns;
-        max_size = max_size * update_columns_size / _tablet_schema->num_columns();
+        int64_t update_data_size = 0;
```

Comment:
Your statistics operation here is quite expensive, and there’s no need to repeat the calculation every time need_flush is called. Move it to the initialization phase of DeltaWriter instead.